storageconnector/gitstorage.go

```diff
@@ -0,0 +1,270 @@
+package storageconnector
+
+import (
+	"strings"
+	"path"
+	"sort"
+	"gopkg.in/yaml.v2"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/core"
+	"io/ioutil"
+	"fmt"
+)
+
+const (
+	featureDir = "features"		// name of the directory where features are located.
+	featureFile = "meta.yml"	// name of the file containing all metadata for a feature.
+	featureSnippet = "Dockerfile"   // the file containing the actual docker snippet.
+	defaultSearchParamsLimit = 100
+)
+
+// yamlFeatureMeta is used for unmarshalling of meta.yml files.
+type yamlFeatureMeta struct {
+	Description string
+	Dependencies []string
+}
+
+// gitStorage is an implementation of StorageReader based on
+// a git repository as storage back-end.
+type gitStorage struct {
+	repo 	*git.Repository
+}
+
+// NewStorageReader returns a StorageReader which uses a public git repository
+// as data source for pazuzu features.
+//
+// url:  The URL to the git repository that serves as data source. The
+//       repository must be publicly accessible.
+//
+// If the repository can't be accessed NewStorageReader returns an error.
+func NewStorageReader(url string) (StorageReader, error) {
+	// OPTIMIZATION: can be an fs repository which is cached and only pulled when needed
+	repo := git.NewMemoryRepository()
+
+	err  := repo.Clone(&git.CloneOptions{
```

Comment:
do you need to Clone each time you create a connection to the repo? Otherwise you will run `Clone` operation each time you execute a command with CLI tool